jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushBlobStep.java

```diff
@@ -83,8 +85,13 @@ public BlobDescriptor call() throws IOException, RegistryException, ExecutionExc
         return blobDescriptor;
       }
 
-      // todo: leverage cross-repository mounts
-      registryClient.pushBlob(blobDescriptor.getDigest(), blob, null);
+      // If base and target images are in the same registry, then use mount/from to try mounting the
+      // BLOB from the base image repository to the target image repository and possibly avoid
+      // having to push the BLOB. See the following for details:
+      // https://docs.docker.com/registry/spec/api/#cross-repository-blob-mount
+      // Must also be checked in {@link AuthenticatePushStep}
+      Optional<String> mountFrom = MountFromSupport.getMountFrom(buildConfiguration);
+      registryClient.pushBlob(blobDescriptor.getDigest(), blob, mountFrom.orElse(null));
```

Comment:
Not sure, but I think it might be preferable to use `isPresent` and `get` over defaulting to `null` when retrieving the value from an `Optional`, so this would be more clearly written as:

```java
if (mountFrom.isPresent()) {
  registryClient.pushBlobWithMount(blobDescriptor.getDigest(), blob, mountFrom.get());
} else {
  registryClient.pushBlob(blobDescriptor.getDigest(), blob);
}
```